go-server/router/router.go

```diff
@@ -2,20 +2,56 @@ package router
 
 import (
 	"go-server/middleware"
+	"net/http"
 
-	"github.com/gorilla/mux"
+	"github.com/gin-gonic/gin"
 )
 
-// Router is exported and used in main.go
-func Router() *mux.Router {
+// CORSMiddleware handles cors headers for security
+func CORSMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
 
-	router := mux.NewRouter()
+		c.Header("Access-Control-Allow-Origin", "http://localhost:3000")
+		c.Header("Access-Control-Allow-Credentials", "true")
+		c.Header("Access-Control-Allow-Headers", "Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, accept, origin, Cache-Control, X-Requested-With")
+		c.Header("Access-Control-Allow-Methods", "POST,HEAD,PATCH,OPTIONS,GET,PUT,DELETE")
+		c.Header("Context-Type", "application/json")
 
-	router.HandleFunc("/api/task", middleware.GetAllTask).Methods("GET", "OPTIONS")
-	router.HandleFunc("/api/task", middleware.CreateTask).Methods("POST", "OPTIONS")
-	router.HandleFunc("/api/task/{id}", middleware.TaskComplete).Methods("PUT", "OPTIONS")
-	router.HandleFunc("/api/undoTask/{id}", middleware.UndoTask).Methods("PUT", "OPTIONS")
-	router.HandleFunc("/api/deleteTask/{id}", middleware.DeleteTask).Methods("DELETE", "OPTIONS")
-	router.HandleFunc("/api/deleteAllTask", middleware.DeleteAllTask).Methods("DELETE", "OPTIONS")
-	return router
+		if c.Request.Method == "OPTIONS" {
+			c.AbortWithStatus(204)
+			return
+		}
+
+		c.Next()
+	}
+}
+
+//Router is exported and used in main.go
+func Router() *gin.Engine {
+	// new apis
+	r := gin.Default()
+
+	r.Use(CORSMiddleware())
+
+	// Setup route group for the API
+	api := r.Group("/api")
+	{
+		api.GET("/", func(c *gin.Context) {
+			c.JSON(http.StatusOK, gin.H{
+				"message": "pong",
+			})
+		})
+	}
+
+	api.GET("/localreps", middleware.LocalRepsHandler)
+	api.GET("/task", middleware.GetAllTask)
+	api.POST("/task", middleware.CreateTask)
+	api.PUT("/task/:id", middleware.TaskComplete)
+	api.PUT("/undoTask/:id", middleware.UndoTask)
+	api.OPTIONS("/deleteTask/:id", middleware.DeleteTask)
```

Comment:
originally had "{id}" (mux syntax) and changed to ":id" -- I usually use keyword parameters, not in-path...